docs/proposals/status-aggregation.md

```diff
@@ -10,39 +10,135 @@ When multiple instances of a resource (e.g. a Gateway) are running across multip
+This is important when initial setup of DNS Records happen as applications start.
+It doesn't make sense to route traffic to a Gateway where a listener isn't ready/attached yet.
+It also comes into play when a Gateway is displaced either due to changing placement decision or removal.
+
+In summary, the IP Addresses and number of attached routes per listener per Gateway instance is needed in the control plane to manage DNS effectively.
+This proposal adds that information the hub Gateway status block.
+This will ensure a decoupling of the DNS logic from the underlying resource/status syncing implementation (i.e. ManifestWork status feedback rules)
+
+First, here are 2 instances of a multi cluster Gateway in 2 separate spoke clusters.
+The yaml is shortened to highlight the status block.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: gateway
+status:
+  addresses:
+  - type: IPAddress
+    value: 172.32.200.0
+  - type: IPAddress
+    value: 172.32.201.0
+  listeners:
+  - attachedRoutes: 0
+    conditions:
+    name: api
+  - attachedRoutes: 1
+    conditions:
+    name: web
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: gateway
+status:
+  addresses:
+  - type: IPAddress
+    value: 172.32.202.0
+  - type: IPAddress
+    value: 172.32.203.0
+  listeners:
+  - attachedRoutes: 1
+    name: api
+  - attachedRoutes: 1
+    name: web
+```
+
+And here is the proposed status aggregation in the hub Gateway:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: gateway
+status:
+  addresses:
+    - type: kuadrant.io/MultiClusterIPAddress
+      value: cluster_1/172.32.200.0
+    - type: kuadrant.io/MultiClusterIPAddress
+      value: cluster_1/172.32.201.0
+    - type: kuadrant.io/MultiClusterIPAddress
+      value: cluster_2/172.32.202.0
+    - type: kuadrant.io/MultiClusterIPAddress
+      value: cluster_2/172.32.203.0
+  listeners:
+    - attachedRoutes: 0
+      name: cluster_1.api
+    - attachedRoutes: 1
+      name: cluster_1.web
+    - attachedRoutes: 1
+      name: cluster_2.api
+    - attachedRoutes: 1
+      name: cluster_2.web
+```
+
+The MultiCluster Gateway Controller will use a custom implementation of the `addresses` and `listenerers` fields.
+The address `type` is of type [AddressType](https://github.com/kubernetes-sigs/gateway-api/blob/f883de997b88dd6ee138930198542da8a9b2f634/apis/v1beta1/shared_types.go#L552), where the type is a domain-prefixed string identifier. The length has a max of 253 characters.
+The value can be split on the forward slash, `/`, to give the cluster name and the underlying Gateway IPAddress value of type IPAddress.
+Only the IPAddress type will be supported at this time, though there is no obvious reason it couldn't be extended to support a Hostname address type later.
```

Comment:
not sure why?

Reply:
Simplest answer is that IPAddress is the only type we've seen so far.
I wouldn't want to add support for something that we don't actively test.

Reply:
We will need HostAddress as well. I suppose at the moment we only use IPAddress, but we will after the proposed DNS record structure changes (being added right now https://github.com/Kuadrant/multicluster-gateway-controller/pull/293) want to support hostnames as well as we will be removing the HostResolver and instead setting a CNAME to the host instead see https://github.com/Kuadrant/multicluster-gateway-controller/blob/main/docs/proposals/DNSRecordStructure.md#geo-agnostic-everything-is-in-a-default--geo-catch-all

Reply:
one thought I had was we should consider dropping kuadrant.io for mgc. IE make our prefix as short  as possible to avoid any issues with long hostnames (edge case I know)

Reply:
> but we will after the proposed DNS record structure changes

Ack.
I'll update the wording to include.
We can add an integration test case as a start.

> make our prefix as short as possible to avoid any issues with long hostnames

The kuadrant.io prefix would only be on the Address type, which would be limited to values of `kuadrant.io/MultiClusterIPAddress` and `kuadrant.io/MultiClusterHostName`.

The length would be a factor for cluster names in the prefix of the address value and the listener name

Reply:
ah yes ofc 👍

Reply:
> one thought I had was we should consider dropping kuadrant.io for mgc. IE make our prefix as short as possible to avoid any issues with long hostnames (edge case I know)

It's a prefix to the address type so will the length not be constant? Using a domain was what it suggests in it's example https://github.com/kubernetes-sigs/gateway-api/blob/f883de997b88dd6ee138930198542da8a9b2f634/apis/v1beta1/shared_types.go#L538